filebeat/prospector/prospector_log.go

```diff
@@ -56,6 +56,21 @@ func (p *ProspectorLog) Run() {
 		p.Prospector.states.Cleanup()
 		logp.Debug("prospector", "Prospector states cleaned up.")
 	}
+
+	// Cleanup of removed files will only happen after next scan. Otherwise it can happen that not all states
+	// were updated before cleanup is called
+	if p.config.CleanRemoved {
+		for _, state := range p.Prospector.states.GetStates() {
+			_, err := os.Stat(state.Source)
+			if err != nil {
+				state.TTL = 0
```

Comment:
setting state.TTL to 0 is pretty scary, given the comparison uses `time.Since(state.timestamp) >= ttl` which is pretty much always true. Even if file is present.